File: nhglib-core/src/io/github/voidzombie/nhglib/graphics/particles/ParticleEffectPool.java
package io.github.voidzombie.nhglib.graphics.particles;

import com.badlogic.gdx.graphics.g3d.particles.ParticleEffect;
import com.badlogic.gdx.utils.Pool;

public class ParticleEffectPool extends Pool<ParticleEffect> {
    private ParticleEffect originalEffect;

    public ParticleEffectPool(ParticleEffect originalEffect) {
        this.originalEffect = originalEffect;
    }

    @Override
    public void free(ParticleEffect object) {
        object.reset();
        super.free(object);
    }

    @Override
    protected ParticleEffect newObject() {
        return originalEffect.copy();
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/graphics/particles/ParticleEffectPool.kt
package io.github.movementspeed.nhglib.graphics.particles

import com.badlogic.gdx.graphics.g3d.particles.ParticleEffect
import com.badlogic.gdx.utils.Pool

class ParticleEffectPool(private val originalEffect: ParticleEffect) : Pool<ParticleEffect>() {

    override fun free(`object`: ParticleEffect) {
        `object`.reset()
        super.free(`object`)
    }

    override fun newObject(): ParticleEffect {
        return originalEffect.copy()
    }
}
